Foundations Slide Deck Content & Notes Update

jordanmccullough opened the pull request:
Tidying up some notes and some command examples.

Commit:
Update Stash chapter example, notes

Commit:
Add Reflog expire all note

Commit:
Add Branch chapter move name

Commit:
Add Log search chapter slide

Commit:
Add Merge chapter slide for checkout options

brntbeer commented on presentations/_posts/foundations/log/0006-01-01-Searching-Commits.md:
```diff
@@ -0,0 +1,15 @@
+tags: ['log']
+---
+
+	git log -S<string-in-patch>
```
Currently already lives in [log additional options](https://github.com/github/teach.github.com/blob/gh-pages/presentations/_posts/foundations/log/0005-01-01-Additional-Options.md)

Recommend moving this first, also don't forget to put a space: `-S <string-in-patch>` vs `-S<string-in-patch>`

matthewmccullough replied:
@brntbeer To your excellent point, I'd like to see it killed (too many things on one slide) on the `log additional options` page

jordanmccullough replied:
@brntbeer Thanks for the note. Oddly, Git plays nice with or without the space between the `-S` and the string. However, for clarity's sake the space definitely makes sense.

brntbeer replied:
Ah. though this is what tripped us up a few months ago when we were teaching down in San jose area together!

matthewmccullough commented on presentations/_posts/foundations/stash/0002-01-01-Stash-Commands.md:
```diff
@@ -7,22 +7,10 @@ categories: ['slidecontent']
-	git stash apply
-
-	# Drop all stashes
-	git stash clear
```
Super happy at this segmenting of stash. I do like to reveal it in pieces like this.

matthewmccullough commented on presentations/_posts/foundations/stash/0003-01-01-Stash-Commands.md:
```diff
@@ -0,0 +1,16 @@
+	git stash pop <name>
+
+	# Merge & keep the latest stash
+	git stash apply
```
I like pop and apply being together on the same slide, as I love to contrast and compare them.

matthewmccullough commented on presentations/_posts/foundations/stash/0005-01-01-Branch-from-a-Stash.md:
```diff
@@ -5,4 +5,4 @@ tags: ['stash']
 
-	$ git branch <branchname> stash@{0}
\ No newline at end of file
+	$ git branch <branchname> <stash-name>
```
Probably just needs some speaker notes about how the format is `stash@{NUM}`

Commit:
Update "What is a Commit" with build-out diagram

Commit:
Add Commit chapter shorthand slides

Commit:
Rename Diff Options slide for title clarity

Commit:
Fix duplicate Log search slide info

Removed duplicate -S option example, added
space for -S option for visual clarity

 # Please enter the commit message for your changes. Lines starting

Commit:
Update RM-MV slide with clearer title, improved diagram

Updated title, omitting the "five states of..." reference
and removed ignore as state of "tracking" in diagram.
Provides for clearer concept within the RM/MV content.

jordanmccullough commented:
Considerably updated slides, fine tuned some diagrams that needed work, added a new build-out for the Commit Structure (commit-->tree-->blob) graphic.

This is near-ready for a merge, if I do say so myself.

matthewmccullough commented:
A quick review says good to go.  What say you @brntbeer since you gave this detailed eyes on the first pass?

brntbeer commented:
ship it. if there's issues we'll fix them as we run through them

jordanmccullough commented:
:ship:ed. Thanks for the review guys.
